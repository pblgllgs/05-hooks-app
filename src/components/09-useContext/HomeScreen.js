import React from 'react'

export const HomeScreen = () => {
    return (
        <>
            <h1>Home Screen</h1>
            <hr />
        </>
    )
}
