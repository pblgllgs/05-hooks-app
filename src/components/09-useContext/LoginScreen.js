import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser} = useContext(UserContext);

    const userlogin = {
        id:123,
        name:'pablo',
        email: 'pablo@gmail.com'
    } 

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className='btn btn-primary' 
                onClick={()=> setUser(userlogin)}
            >
                Login
            </button>
        </div>
    )
}
