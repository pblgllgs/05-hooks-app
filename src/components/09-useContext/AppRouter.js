import React from 'react'

import {
    BrowserRouter as Router,
    Switch,
    Route,
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <div>
            <Router>
                <NavBar />
                <Switch>
                    <Route exact path="/" element={<HomeScreen />} />
                    <Route exact path="/about" element={<AboutScreen />} />
                    <Route exact path="/login" element={<LoginScreen />} />
                    <Route path="*" element={<HomeScreen />} />
                </Switch>
            </Router>
        </div>
    )
}