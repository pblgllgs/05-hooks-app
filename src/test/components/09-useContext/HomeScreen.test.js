import { mount } from 'enzyme'
import { HomeScreen } from '../../../components/09-useContext/HomeScreen'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('pruebas en <HomeScreen />', () => {
    //se crea un user para usar en el test
    const user = {
        name:'pablo',
        email: 'pblgllgs@gmail.com'
    }
    //capturamos el componente montado
    const wrapper = mount(
        //se agrega un contexto para la prueba y se utiliza el valor de usuario creado
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );
    test('debe de mostrarse correctamente', () => {
        //coomparamos con un snapshot
        expect(wrapper).toMatchSnapshot()
    })
})
