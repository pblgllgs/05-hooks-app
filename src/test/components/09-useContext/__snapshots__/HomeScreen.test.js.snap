// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <HomeScreen /> debe de mostrarse correctamente 1`] = `
Array [
  <h1>
    Home Screen
  </h1>,
  <hr />,
  <pre
    className="container"
  >
    {
   "name": "pablo",
   "email": "pblgllgs@gmail.com"
}
  </pre>,
]
`;
